static/app/views/releases/utils/useDrawerFlags.tsx

```diff
@@ -0,0 +1,80 @@
+import type {Event} from 'sentry/types/event';
+import {useApiQuery} from 'sentry/utils/queryClient';
+import useOrganization from 'sentry/utils/useOrganization';
+import {useOrganizationFlagLog} from 'sentry/views/issueDetails/streamline/hooks/featureFlags/useOrganizationFlagLog';
+import {useGroup} from 'sentry/views/issueDetails/useGroup';
+
+interface Params {
+  enabled: boolean;
+  eventId: string | undefined;
+  groupId: string | undefined;
+  query: Record<string, any>;
+}
+
+/**
+ * Returns the feature flags that have been changed in a given time period and that appear on the event.
+ */
+export function useDrawerFlags({eventId, groupId, query, enabled}: Params) {
+  const organization = useOrganization();
+  const {
+    data: group,
+    isPending: isGroupPending,
+    isError: isGroupError,
+    error: groupError,
+  } = useGroup({
+    groupId: groupId!,
+    options: {enabled: enabled && Boolean(groupId && eventId)},
+  });
+
+  const projectSlug = group?.project.slug;
+  const {
+    data: event,
+    isPending: isEventPending,
+    isError: isEventError,
+    error: eventError,
+  } = useApiQuery<Event>(
+    [`/organizations/${organization.slug}/events/${projectSlug}:${eventId}/`],
+    {
+      staleTime: Infinity,
+      enabled: enabled && Boolean(eventId && projectSlug && organization.slug),
+    }
+  );
+
+  const eventFlags = event?.contexts?.flags?.values?.map(f => f.flag);
+
+  const {
+    data: rawFlagData,
+    getResponseHeader,
+    isPending: isFlagsPending,
+    isError: isFlagsError,
+    error: flagsError,
+  } = useOrganizationFlagLog({
+    organization,
+    query: {
+      ...query,
+      flag: eventFlags,
```

Comment:
So this is different than `useIntersectionFlags` because we want to query w/ the event's flags so that the results can be properly paginated (vs finding the intersection on the FE). This could be an issue if the event has a lot of feature flags